README.md

```diff
@@ -1,5 +1,146 @@
+        int stackTop = stack.top();
+
+        // Assert
+        assertThat(emptyStackTop, equalTo(value));
+        assertThat(stackTop, equalTo(value));
+REPOBEE-SANITIZER-REPLACE-WITH
+        fail("Not implemented");
+REPOBEE-SANITIZER-END
+    }
+}
+````
+> Example 2: The code is the same as for example 1, but we have added a `REPOBEE-SANITIZER-REPLACE-WITH` marker.
+
+As we can see in Example 2, this lets us provide two versions of a function, one that is current, and one that will replace it. Example 1 and 2 shows us a piece of code used in the KTH course DD1338. This code is part of an assignment where students are asked to implement a test function. The example shows a finished solution that is availiable to the teachers of the course. However, because of `repobee-sanitizer` and the `REPLACE-WITH` marker, the code can be reduced to the following:
+
+````
+class StackTest {
+    @Test
+    public void topIsLastPushedValue() {
+        fail("Not implemented");
+    }
+}
+````
+
+> Example 3: Skeletonized code that is provided to students.
+
+We can see that the only code that remains inside the function is that of the `REPLACE-WITH` marker. This gives us the main usage that `repobee-santizer` was developed for, it allows us to combine finished solutions with the "skeletonized" solutions that are provided to students.
+
+# Prefixing
+Sometimes (usually) we want code that can run, its a good thing then that `repobee-sanitizer` blocks can be commented out! Example two will produce the same output as the following:
+
+````
+class StackTest {
+    @Test
+    public void topIsLastPushedValue() {
+//REPOBEE-SANITIZER-BLOCK
+        // Arrange
+        int value = 1338;
+
+        // Act
+        emptyStack.push(value);
+        stack.push(value);
+
+        int emptyStackTop = emptyStack.top();
+        int stackTop = stack.top();
+
+        // Assert
+        assertThat(emptyStackTop, equalTo(value));
+        assertThat(stackTop, equalTo(value));
+//REPOBEE-SANITIZER-REPLACE-WITH
+//        fail("Not implemented");
+//REPOBEE-SANITIZER-END
+    }
+}
+````
+> Example 4: Java code with `repobee-sanitizer` related syntax commented out 
+
+`repobee-sanitizer` automatically detects if there is a prefix in front of any markers. This way we can have java comments: `//`, python comments: `#` or similar preceding our markers. **This means code can still compile!**
+
+## Some rules for prefixing:
+
+`repobee-sanitizer`:
+* Determines prefix as any text that comes before `REPOBEE-SANITIZER-BLOCK`
+* Only determines prefix on a block-to-block basis, meaning that the prefix selected at a `BLOCK` marker must be used untill and including the next `END` marker 
+    * This means that all `repobee-sanitizer` blocks can have individuall prefixes
+
+
+# Syntax
+for `repobee-sanitizer` to work, marker syntax must be correct, this includes spelling of the markers themselves, the markers currently are as follows:
+
+- REPOBEE-SANITIZER-BLOCK
```

Comment:
Looking at this, I think we should probably rename the start marker to `REPOBEE-SANITIZER-START`. But let's leave that for later.